src/simulation/m_riemann_solvers.fpp

```diff
@@ -1387,6 +1396,15 @@ contains
                                 xi_M = (5d-1 + sign(5d-1, s_S))
                                 xi_P = (5d-1 - sign(5d-1, s_S))
 
+                                if (low_Mach == 1) then
+                                    pcorr = rho_L*rho_R* &
```

Comment:
these `if(low_mach = 1` (or `2`) routines seem to be repeated for all of the types of Riemann solver. they should be turned into their own functions or subroutines.

Reply:
I move them into a subroutine